bundles/AdminBundle/Controller/Admin/PortalController.php

```diff
@@ -291,24 +294,24 @@ public function portletModifiedAssetsAction(Request $request)
             'limit' => 10,
             'order' => 'DESC',
             'orderKey' => 'modificationDate',
-            'condition' => "userModification = '".$this->getAdminUser()->getId()."'",
         ]);
 
+        $user = Admin::getCurrentUser();
+        $list->filterAccessibleByUser($user, 'view');
+
         $response = [];
         $response['assets'] = [];
 
         foreach ($list as $doc) {
             /**
              * @var Asset $doc
              */
-            if ($doc->isAllowed('view')) {
-                $response['assets'][] = [
-                    'id' => $doc->getId(),
-                    'type' => $doc->getType(),
-                    'path' => $doc->getRealFullPath(),
-                    'date' => $doc->getModificationDate(),
-                ];
-            }
+            $response['assets'][] = [
```

Comment:
shouldn't we keep that check for extra security? we also have them in the tree controllers...

Reply:
No, `isAllowed()` executes exactly the same database query again:
Compare
https://github.com/pimcore/pimcore/blob/86ab63e336ce5d36201370381b3aaf177e56b435/models/Asset/Dao.php#L429-L449
with https://github.com/pimcore/pimcore/blob/86ab63e336ce5d36201370381b3aaf177e56b435/models/Asset/Listing.php#L85-L101
But of course then for every item in the loop! So beside being slow this is also redundant.

These `filterAccessibleByUser` methods in the Listings have potential to be used for all permission checks in the backend within the actual data-fetching query. No additional `isAllowed()` calls in loops necessary anymore. This will result in big performance improvements...

Reply:
Again sorry for the pickiness, but that is not the case. 

The query of `filterAccessibleByUser` compares based on path, the `isAllowed` compares based on ids. There are edge cases, where that matters ... e.g. constellations like 
![image](https://user-images.githubusercontent.com/8792145/121653716-47e81380-ca9d-11eb-9471-2bf39036947f.png)

These constellations are filtered with the `isAllowed` call - also in the trees and grids, see for example [here](https://github.com/pimcore/pimcore/blob/092422059721cb76426194c60cddab433ce1d3c2/bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php#L216) or [here](https://github.com/pimcore/pimcore/blob/092422059721cb76426194c60cddab433ce1d3c2/bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php#L1814).

Besides that, `isAllowed` fires an `AdminEvents::ELEMENT_PERMISSION_IS_ALLOWED` event - which of course needs to be fired in the dashboard case too. 

So it is about consistency and security. Permissions is a very delectat topic and thus we have to be very careful there. 

In general, there is room for improvement for the whole permission implementation, especially when it comes to performance, see also https://github.com/pimcore/pimcore/issues/6771 ... but still, what we change/improve must not introduce any regressions.

Reply:
Ok, I added the `isAllowed()`. Perhaps we will find a correct way in #6771 which also enhances performance.